Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
-We distinguish between user validation, performed by the user (through a user interface), and machine validation, that can be done without user interaction.
+This message introduces the `contract_id` to identify the contract. It's derived from the funding transaction by combining the `funding_txid` and the `funding_output_index`, using big-endian exclusive-OR (i.e. `funding_output_index` alters the last 2 bytes).
 
-### Offer Validation
+1. type: ??? (`sign_dlc`)
+2. data:
+   * [`contract_id`:`contract_id`]
+   * [`cet_signatures`:`cet_signatures`]
+   * [`signature`:`refund_signature`]
+   * [`funding_signatures`:`funding_signatures`]
 
-#### User validation
-* Contract Information: confirm that the contract information matches what was agreed upon.
-* Oracle information: confirm that the oracle specified in the message matches what was agreed upon.
-* A fund amount: confirm that the input amount matches what was agreed upon.
-* Estimatesmartfee: confirm that it corresponds to what was agreed upon.
-* CET CSV delay: confirm that it corresponds to what was agreed upon.
-* Refund locktime: confirm that it corresponds to what was agreed upon.
+#### Requirements
 
-#### Machine validation
+The sender MUST:
 
-* A fund amount: given this and B fund amount, need to validate that the amount incorporates the required fees for the CET or refund transaction as well as the closing transaction, and that the total input amount is greater than the payout amounts plus the fees to be paid.
-* A public key: validate that the public key is well formed.
-* Alice inputs: validate that the referenced outputs are in fact unspent and are either P2WSH or P2WPKH.
+  - set `channel_id` by exclusive-OR of the `funding_txid` and the `funding_output_index` from the `offer_dlc` and `accept_dlc` messages.
+  - set `cet_signatures` to valid adaptor signatures, using its `funding_pubkey` for each CET, as defined in the [transaction specification](Transactions.md#contract-execution-transaction) and using signature public keys computed using the `offer_dlc`'s `contract_info` and `oracle_info` as adaptor points.
+  - include an adaptor signature in `cet_signatures` for every event specified in the `offer_dlc`'s `contract_info`.
+  - set `refund_signature` to the valid signature, using its `funding_pubkey` for the refund transaction, as defined in the [transaction specification](Transactions.md#refund-transaction).
+  - set `funding_signatures` to valid input signatures.
+  - include a signature in `funding_signatures` for every funding input specified in the`offer_dlc` message.
 
-### Accept validation
+The recipient:
 
-#### User validation
-* B fund amount: confirm that the fund amount matches what was agreed upon.
+  - if any `signature` is incorrect:
+    - MUST reject the contract.
+  - MUST NOT broadcast the funding transaction before receipt of a valid `sign_dlc`.
+  - on receipt of a valid `sign_dlc`:
+    - SHOULD broadcast the funding transaction.
 
```

Comment:
Maybe add something about if the Initiator doesn't see the funding tx in a timely manner, they can exit the DLC by spending one of their funding inputs.

Reply:
I agree that should be specified but I believe it belongs in its own doc similar to [BOLT 5](https://github.com/lightningnetwork/lightning-rfc/blob/master/05-onchain.md)

Reply:
Should also include fee-bumping and such in that doc